### PR TITLE
Provide more detail to the output of --help

### PR DESCRIPTION
**Pull request by JessRudder**
Closes https://github.com/github/models/issues/321

This PR updates the `--help` output for the following:

`gh models --help`
`gh models list --help`
`gh models view --help`
`gh models run --help`

In order to make it easier to format the longer string I've included the heredoc module.

The contents that I wrote are a rough draft based on my basic understanding of how things work and what's helpful to include in the docs. Please provide feedback on the content!

Some specific points I'd like feedback on (though all feedback is welcome):
1) Is the wording clear and concise?
2) ~~Should we mention Azure at all (also, am I even correct that this is coming from Azure)?~~ Consensus is no.
3) Are there rate limits/limitations on certain commands that I should include in the output?
4) ~~I followed some other gh CLIs and mentioned https://cli.github.com/manual/gh_models - will we actually have docs there (like [codespace](https://cli.github.com/manual/gh_codespace) and other extensions do)?~~ Using the readme for the extension for now, can change later if we end up in the online manual.

### CODE REVIEW

File: cmd/view/view.go [hunk 1]
```diff
@@ -15,6 +16,16 @@ func NewViewCommand(cfg *command.Config) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "view [model]",
 		Short: "View details about a model",
+		Long: heredoc.Docf(`
+			Makes an HTTP request to the Azure API for details on the selected model and prints the response.
```
Comment (cschleiden) [inline cmd/view/view.go:20]: Same here, just tell them that this returns details about the specified model?

File: cmd/root.go [hunk 2]
```diff
@@ -21,6 +22,16 @@ func NewRootCommand() *cobra.Command {
...
 		Short: "GitHub Models extension",
+		Long: heredoc.Docf(`
+			GitHub Models CLI extension allows you to experiment with AI models from the command line.
+
+			To see a list of all available commands, run %[1]sgh models help%[1]s. To run the extension in
+			interactive mode, run %[1]sgh models run%[1]s. This will prompt you to select a model and then
+			to enter a prompt. The extension will then return a response from the model.
+
+			For more information about what you can do with GitHub Models extension see the manual
+			at https://cli.github.com/manual/gh_models
```
Comment (cheshire137) [inline cmd/root.go:33]: Guessing this URL is aspirational. It currently 404s; wondering if we should omit the URL from the output till the page exists. Could use the readme URL in the meantime: https://github.com/github/gh-models/blob/main/README.md.

File: cmd/root.go [hunk 3]
```diff
@@ -21,6 +22,16 @@ func NewRootCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "models",
 		Short: "GitHub Models extension",
+		Long: heredoc.Docf(`
+			GitHub Models CLI extension allows you to experiment with AI models from the command line.
+
+			To see a list of all available commands, run %[1]sgh models help%[1]s. To run the extension in
```
[the remark this answers is not in the document]
Comment (JessRudder) [reply cmd/root.go:28]: It does! Not sure if that's the best/only way to do it, but, that's how it was done in the api extension that was mentioned in the issue.

**Review by cschleiden — APPROVED** [review]
LGTM

**Pull request merged**